### PR TITLE
BorderRadius RTL

### PR DESCRIPTION
**Pull request by Hixie**
Add an RTL version of BorderRadius.
Minor fixes to EdgeInsets to match.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth

File: packages/flutter/lib/src/painting/border_radius.dart [hunk 1]
```diff
@@ -6,152 +6,842 @@ import 'package:flutter/foundation.dart';
 
 import 'basic_types.dart';
 
+/// Base class for [BorderRadius] that allows for text-direction aware resolution.
+///
+/// A property or argument of this type accepts classes created either with [new
+/// BorderRadius.only] and its variants, or [new BorderRadiusDirectional.only]
```
Comment (abarth) [inline packages/flutter/lib/src/painting/border_radius.dart]: Does this line break mess up the docs?  It messes up GitHub's colorization.

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/painting/border_radius.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/painting/border_radius.dart]: Pretty sure dartdoc handles this fine, but I'll try to remember to check.

Comment (Hixie) [reply packages/flutter/lib/src/painting/border_radius.dart]: yup, looks fine

**Pull request merged**